set_tags/utils.py

```diff
@@ -56,6 +59,38 @@ def get_synapse_owner_id(tags):
   else:
     raise ValueError(f'Expected to find {principal_arn_tag} in {tags}')
 
+def get_synapse_owner_id(tags):
+  '''Find the value of the principal ARN among the resource tags. The principal
+  ARN tag is applied by AWS and it's value should be in the following format
+  'arn:aws:sts::111111111:assumed-role/ServiceCatalogEndusers/378505'
+  :param tags: resource tags can take two forms
+    * A list of dictionary of key/value pairs
+      i.e. tags:{'Key':'string', 'Value':'string'}
+    * A dictionary of key pairs
+      i.e. tags:{'string':'string'}
+  returns: the synapse user id (i.e. 378505)
+  '''
+  principal_arn_tag = 'aws:servicecatalog:provisioningPrincipalArn'
+  synapse_owner_id = None
+
+  if isinstance(tags, list):  # tags:{'Key':'string', 'Value':'string'}
+    for tag in tags:
+      if tag.get('Key') == principal_arn_tag:
+        principal_arn_value = tag.get('Value')
+        synapse_owner_id = principal_arn_value.split('/')[-1]
+  if isinstance(tags, dict):  # tags:{'string':'string'}
+    for key, value in tags.items():
```

Comment:
Why not just replace the for loop with:
```
if principal_arn_tag in tags:
     principal_arn_value = tags.get(principal_arn_tag):
     synapse_owner_id = principal_arn_value.split('/')[-1]

```
?

Reply:
because we can have multiple tags in the dict that we need to iterate thru.  take a look at `test_dict_tag_present`, the dict of tags can be this..
```
    tags = {
      'heresatag': 'heresatagvalue',
      'theresatag':'theresatagvalue',
      'aws:servicecatalog:provisioningPrincipalArn':'foo/bar'
    }
```

Reply:
>  we can have multiple tags in the dict that we need to iterate thru.
I don't think that's true:  There is at most ONE tag in the dict having key 'aws:servicecatalog:provisioningPrincipalArn' (as in your `test_dict_tag_present` example).

Reply:
ahh of course, good suggestion.  thanks!